test/controllers/surveys_controller_test.rb

```diff
@@ -53,21 +79,35 @@ class SurveysControllerTest < ActionDispatch::IntegrationTest
   end
 
   test "should show survey" do
-    get survey_url(@survey)
-    assert_response :success
+    @survey = create(:survey, user: @user, published: true)
+
+    sign_in @user
+    get survey_path(@survey)
+    assert_redirected_to survey_path(@survey)
```

Comment:
We should use something other than redirected to here as we're probably not hitting a redirect